### PR TITLE
devp: Migrate Jenkinsfile to Actions

### PR DESCRIPTION
**Pull request by markpitchless**
See fac/dev-platform#35

### CODE REVIEW

**Comment by markpitchless** [comment]
The push is failing but the error is not getting raised. Looks like bug in gem action. Also needs to push to rubygems (where does the key come from?)

File: .github/workflows/freeagent-gem.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    - name: Release
+      if:   github.ref == 'refs/heads/master'
+      uses: fac/ruby-gem-push-action@v2
+      with:
+        key: github
+
+    # PR branch builds will release pre-release gems
+    - name: Pre-Release
+      if:   github.ref != 'refs/heads/master'
+      uses: fac/ruby-gem-push-action@v2
```
Comment (robsimpsondev) [inline .github/workflows/freeagent-gem.yml:53]: @markpitchless looks like this step failed--is that something you're planning to look at?

Comment (markpitchless) [reply .github/workflows/freeagent-gem.yml:53]: @robsimpsondev it is, just fixing that now, I missed that this is supposed to push to rubygems.

Comment (robsimpsondev) [reply .github/workflows/freeagent-gem.yml:53]: Looks like it failed again with the following failure mode:
```
Pushing gem to https://rubygems.org...
Access Denied. Please sign up for an account at https://rubygems.org
```
Should this be a failure mode (i.e. cause the check to fail)? It's green at the mo!

Comment (markpitchless) [reply .github/workflows/freeagent-gem.yml:53]: Trying to fix that as well, it should have failed. grrr.

Comment (markpitchless) [reply .github/workflows/freeagent-gem.yml:55]: @robsimpsondev had a bit of fun getting the rubygem push working, fixed now. Better error reporting to come in future release of the action ;-)

Comment (robsimpsondev) [reply .github/workflows/freeagent-gem.yml:55]: image
There we go!

File: lib/hermod/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module Hermod
-  VERSION = "2.6.1"
```
Comment (robsimpsondev) [inline lib/hermod/version.rb:2]: Think we need to set the version back to 2.6.1 for the merge to master?

Comment (robsimpsondev) [reply lib/hermod/version.rb:2]: And then yank the 2.6.2.pre2 version from RubyGems? 🤔

**Review by robsimpsondev — CHANGES_REQUESTED** [review]

File: lib/hermod/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
...
+  VERSION = "2.6.2.pre2"
```
Comment (dgholz) [inline lib/hermod/version.rb:2]:
```suggestion
  VERSION = "2.6.1"
```

**Review by robsimpsondev — APPROVED** [review]

**Pull request merged**